### ISSUE TITLE
404 with localeSubpaths

### ISSUE THREAD

**Issue by Uniiq** [isaachinman/next-i18next#790]
## Describe the bug
I get a 404 with localeSubpaths only on production build. In develop everthing works fine.

## Occurs in next-i18next version
v6.0.2

## Steps to reproduce
I get no error so i have no clue

## Expected behaviour
http://localhost:3000/en -> statusCode 200 and not 404

## OS (please complete the following information)
 - Browser: Chrome: 84.0.4147.105

## Additional context
- next.js v9.5.1

I will try to submit a reproducible repo.

**Comment by isaachinman** [comment]
Let me know when you put together a reproducible repo, and I'll have a look.

**Comment by knyuwork** [comment]
I have the same issue. 

When I first com to http://localhost:3000/en, it loads the english. 
However, once I change the language, i got 404 for the common.json need to be used for the new language

e.g. `http://localhost:3000/zh-Hant/src/utils/i18n/locales/zh-Hant/common.json` 404 (Not Found)

**Comment by michelalbers** [comment]
Got the same problem. Reproducible Repo: https://github.com/michelalbers/next-i18next-404

**Comment by isaachinman** [comment]
@michelalbers Couple issues with that repo regarding your TS setup. Regardless, I was able to run the app via `yarn build && yarn start`, and can not see any 404 errors. Can you provide specific instructions as to how I can reproduce the error?

**Comment by michelalbers** [comment]
Try yarn dev :)

**Comment by isaachinman** [comment]
Running with `yarn dev` does not produce any 404s for me either.

**Comment by michelalbers** [comment]
I have the feeling it doesn’t work when u didn’t run next build initially. Will confirm when I’m back at the office.

**Comment by isaachinman** [comment]
I did a fresh clone of the repo, and then just did `yarn && yarn dev`. I had to install TS deps because they were not present. I did not run `yarn build`.

**Comment by michelalbers** [comment]
Figured it out ... I had next.js 9.4.4 in my cache which does not support rewrites. Next.js 9.5 is required which came out 11 days ago. Could we add that to the readme?

**Comment by gpessa** [comment]
How did you fix it exactly? How can i clean the cache?

**Comment by michelalbers** [comment]
> How did you fix it exactly? How can i clean the cache?

In my case it was as simple as ```yarn add next@^9.5.0```

**Comment by isaachinman** [comment]
Well, [it's in the `package.json`](https://github.com/isaachinman/next-i18next/blob/master/package.json#L116), and you would have been receiving warnings from npm or yarn. Not sure it's worth adding to docs.

**Comment by gpessa** [comment]
I still get a 404

**Comment by michelalbers** [comment]
> Well, [it's in the `package.json`](https://github.com/isaachinman/next-i18next/blob/master/package.json#L116), and you would have been receiving warnings from npm or yarn. Not sure it's worth adding to docs.

Yes you get a peer dependecy warning ... but in every bigger project you might have quite a few from them. Anyway - my issue is resolved. Thank you very much @isaachinman

**Comment by isaachinman** [comment]
@gpessa and @Uniiq you will need to provide a reproducible repository if you would like help debugging.

**Comment by Uniiq** [comment]
@isaachinman i sry for the late response. The first problem was i forgot to remove `await nextI18next.initPromise;` from custom next server file (i upgraded from `next-i18next@4.5.0` ).The second problem is i use `cacheable-response` to cache the SSR response and somehow `next-i18next` interfere with my caching function and redirect me to 404 error page (caching is only enabled on production). 

In short there is a bug on my side. :)

**Comment by anialamo** [comment]
> @isaachinman i sry for the late response. The first problem was i forgot to remove `await nextI18next.initPromise;` from custom next server file (i upgraded from `next-i18next@4.5.0` ).The second problem is i use `cacheable-response` to cache the SSR response and somehow `next-i18next` interfere with my caching function and redirect me to 404 error page (caching is only enabled on production).

@Uniiq sorry! what is the solution for this bug? This only happens in production mode. What should I do? Some help please?
